src/NServiceBus.Transport.RabbitMQ/RabbitMQTransportInfrastructure.cs

```diff
@@ -122,13 +122,23 @@ public override Task Stop()
 
         IRoutingTopology CreateRoutingTopology()
         {
-            var durable = settings.DurableMessagesEnabled();
-
             if (!settings.TryGet(out Func<bool, IRoutingTopology> topologyFactory))
             {
                 throw new InvalidOperationException("A routing topology factory has not been configured. Configure a routing topology with a TransportExtensions<RabbitMQTransport>.UseXXXRoutingTopology() method.");
             }
 
+            if (!settings.TryGet(SettingsKeys.UseDurableExchangesAndQueues, out bool durable))
+            {
+                if (settings.DurableMessagesEnabled())
+                {
+                    durable = true;
+                }
+                else
+                {
+                    throw new Exception("When durable messages are disabled, 'EndpointConfiguration.UseTransport<RabbitMQTransport>().UseDurableExchangesAndQueues()' needs to be called to specify exchange and queue durability settings.");
+                }
+            }
+
```

Comment:
For some reason (it could be just me) I'm still finding this a little difficult to read. I think it's the name `durable` and polarity of the inner conditional. How would this look instead?

```C#
if (!settings.TryGet(SettingsKeys.UseDurableExchangesAndQueues, out bool useDurableExchangesAndQueues))
{
    if (!settings.DurableMessagesEnabled())
    {
        throw new Exception(...);        
    }
    
    useDurableExchangesAndQueues = true;
}
```

Reply:
I wrote it originally to avoid the inner `!`, but I can see how "if not enabled, throw" might be clearer. I'll make the change.